### PR TITLE
[type: feature #2451] support regsiter instance to consul

### PR DESCRIPTION
**Pull request by babynata**
[type: feature #2451] 

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-register-center/shenyu-register-instance/shenyu-register-instance-consul/pom.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.apache.shenyu</groupId>
+        <artifactId>shenyu-register-instance</artifactId>
+        <version>2.4.2-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>shenyu-register-instance-consule</artifactId>
```
Comment (yu199195) [inline shenyu-register-center/shenyu-register-instance/shenyu-register-instance-consul/pom.xml:28]: consule --》 consul

File: shenyu-register-center/shenyu-register-instance/shenyu-register-instance-consul/src/main/java/org/apache/shenyu/register/instance/consul/ConsulInstanceRegisterRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        String realNode = RegisterPathConstants.buildRealNode(instancePath, instanceNodeName);
+        String nodeData = GsonUtils.getInstance().toJson(instance);
+        consulClient.setKVValue(realNode, nodeData);
+
+        LOGGER.info("consul client register success: {}", nodeData);
+    }
+
+    @Override
+    public void close() {
+
```
Comment (yu199195) [inline shenyu-register-center/shenyu-register-instance/shenyu-register-instance-consul/src/main/java/org/apache/shenyu/register/instance/consul/ConsulInstanceRegisterRepositoryImpl.java:74]: used consulClient.close

File: shenyu-register-center/shenyu-register-instance/shenyu-register-instance-consul/src/main/java/org/apache/shenyu/register/instance/consul/ConsulInstanceRegisterRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        consulClient.agentServiceRegister(this.service);
+    }
+
+    @Override
+    public void persistInstance(InstanceRegisterDTO instance) {
+        String instanceNodeName = buildInstanceNodeName(instance);
+        String instancePath = RegisterPathConstants.buildInstanceParentPath();
+        String realNode = RegisterPathConstants.buildRealNode(instancePath, instanceNodeName);
+        String nodeData = GsonUtils.getInstance().toJson(instance);
+        consulClient.setKVValue(realNode, nodeData);
```
Comment (yu199195) [inline shenyu-register-center/shenyu-register-instance/shenyu-register-instance-consul/src/main/java/org/apache/shenyu/register/instance/consul/ConsulInstanceRegisterRepositoryImpl.java:67]: pls used consulClient.agentServiceRegister API

File: shenyu-register-center/shenyu-register-instance/shenyu-register-instance-consul/src/main/java/org/apache/shenyu/register/instance/consul/ConsulInstanceRegisterRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
+package org.apache.shenyu.register.instance.consul;
```
Comment (yu199195) [inline shenyu-register-center/shenyu-register-instance/shenyu-register-instance-consul/src/main/java/org/apache/shenyu/register/instance/consul/ConsulInstanceRegisterRepositoryImpl.java:1]: add license  apache 2.0

**Comment by midnight2104** [comment]
hi~ @babynata  can you add license apache 2.0 to  `ConsulInstanceRegisterRepositoryImpl.java` ?

**Pull request closed without merging**